### PR TITLE
Fix for #68

### PR DESCRIPTION
**Pull request by torfjelde**
Fix for #68

TODO: add test

### CODE REVIEW

File: src/filldist.jl [hunk 1]
```diff
@@ -65,6 +65,10 @@ function _flat_logpdf_mat(dist, x)
     end
 end
 
+function Distributions.rand(rng::Random.AbstractRNG, d::FillVectorOfUnivariate)
```
Comment (mohamed82008) [inline src/filldist.jl:68]: This should be covered by `Distributions.Product`.

Comment (mohamed82008) [reply src/filldist.jl:68]: But of course it is not. So happy to merge this.

Comment (torfjelde) [reply src/filldist.jl:68]: Yeah, so last night I didn't get have the time to really look into it, but it seems like it's actually a bug with how `Fill` is handled by `broadcast!`:
```julia
dist = dist = filldist(Gamma(), 2)
x = zeros(2)
broadcast!(dn -> rand(dn), x, dist.v)
```
results in the same issue. Going down a rabit-hole, I think the issue is how `materialize!` (https://github.com/JuliaLang/julia/blob/0bab06f8a7f70d4518ad30e19a4c0a1968c875ad/base/broadcast.jl#L823) acts, which means there's something FillArrays.jl's broadcasting style I think. I'll open an issue over at FillArrays.jl

Comment (torfjelde) [reply src/filldist.jl:68]: Seems like good ole' @willtebbutt as been down this road before: https://github.com/JuliaArrays/FillArrays.jl/issues/29#issue-359096800

**Pull request merged**